### PR TITLE
[EXE-1528] Send jdbcOptions as MergedParameter to driver

### PR DESCRIPTION
**Pull request by ShaoFuWu**
https://actioniq.atlassian.net/browse/EXE-1528
Didn't send queryGroup to driver, details [here](https://github.com/ActionIQ/aiq/pull/45592#issuecomment-1526308915)
```
sc.setLogLevel("info")
val redshiftOpts = Map(
  "aws_iam_role" -> "arn:aws:iam::938824509083:role/journeys-redshift-dev-redshift-role",
  "tempDir" -> s"s3a://aiq-1d-expire-dev/redshift/109",
  "driver" -> "com.amazon.redshift.jdbc42.Driver",
  "dbtable" -> "qahybridredshift.rd_dim_item",
  "queryGroup" -> "123456678908",
  "aiq_partner" -> "partnerName",
  "ApplicationName" -> "appName",
  "password" -> "",
  "user" -> "admin",
  "url" -> "jdbc:redshift://test-hybridcompute.938824509083.us-east-1.redshift-serverless.amazonaws.com:5439/dev?"
)
spark.read.format("io.github.spark_redshift_community.spark.redshift").options(redshiftOpts).load().show()
```

Sending a query from spark-shell and see the query_label `select * from SYS_QUERY_HISTORY where query_label = '123456678908'

### Deploy Note
```
sbt publishM2       
aws s3 sync ~/.m2/repository/io/github/spark-redshift-community/ s3://aiq-artifacts/releases/io/github/spark-redshift-community
```

### CODE REVIEW

File: src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCWrapper.scala [hunk 1]
```diff
@@ -225,9 +224,9 @@ private[redshift] class JDBCWrapper {
       throw new IllegalArgumentException(s"Did not find registered driver with class $driverClass")
     }
     val properties = new Properties()
-    credentials.foreach { case(user, password) =>
-      properties.setProperty("user", user)
-      properties.setProperty("password", password)
+    mergedParameters.parameters.foreach { case (key, value) =>
+      log.info(s"key: ${key} value ${value}")
+      properties.setProperty(key, value)
```
Comment (yannistze) [inline src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCWrapper.scala]: Here you will log every key value, not just the `user`, `password` ones like in the original implementation 🤔 Do you need to be logging them for "Prod" or is this a testing remnant ? I am not sure we should be logging them cause they may contain creds 🤷‍♂️

Comment (ShaoFuWu) [reply src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftJDBCWrapper.scala]: fuuuuuuck will remove, this is for my testing 🤦, good catch!

**Review by yannistze — APPROVED** [review]
Lgtm 👍

File: src/main/scala/io/github/spark_redshift_community/spark/redshift/DefaultSource.scala [hunk 2]
```diff
@@ -97,7 +97,7 @@ class DefaultSource(
     }
 
     def tableExists: Boolean = {
-      val conn = jdbcWrapper.getConnector(params.jdbcDriver, params.jdbcUrl, params.credentials)
```
Comment (MasterDDT) [inline src/main/scala/io/github/spark_redshift_community/spark/redshift/DefaultSource.scala:100]: I'm curious, did you look into the history of why they ignore all the other params?

Even on latest master, only use these 3 params and ignore the others https://github.com/spark-redshift-community/spark-redshift/blob/master/src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftRelation.scala#L111

**Comment by MasterDDT** [comment]
Nice 👍 , this looks like an easy one you could contribute back to the open source fork?

File: src/main/scala/io/github/spark_redshift_community/spark/redshift/DefaultSource.scala (hunk 2 above)
Comment (ShaoFuWu) [reply src/main/scala/io/github/spark_redshift_community/spark/redshift/DefaultSource.scala:100]: Looks like they were fixing the same `invalid character` 8 years ago like we were last week 😅 , the way they fix is passing user/password as part of the options in this [pr](https://github.com/databricks/spark-redshift/issues/132#issuecomment-174050108)

**Pull request merged**